File: Lab_2/task_3/3.php
<?php

// Функция для переворачивания числа и удаления ведущих нулей
function reverseNumber($num) {
    $reversed = 0;
    while ($num > 0) {
        $reversed = $reversed * 10 + $num % 10;
        $num = intdiv($num, 10);
    }
    return $reversed;
}

echo "Введите количество чисел: ";
$n = (int)fgets(STDIN);

echo "Введите $n чисел: ";
for ($i = 0; $i < $n; ++$i) {
    $num = (int)fgets(STDIN);
    echo reverseNumber($num) . " ";
}

echo PHP_EOL;

?>

File: Lab_2/task_3/3.cpp
#include <iostream>

using namespace std;

// Функция для переворачивания числа и удаления ведущих нулей
int reverseNumber(int num) {
    int reversed = 0;
    while (num > 0) {
        reversed = reversed * 10 + num % 10;
        num /= 10;
    }
    return reversed;
}

int main() {
    int n;
    cout << "Введите количество чисел: ";
    cin >> n;

    cout << "Введите " << n << " чисел: ";
    for (int i = 0; i < n; ++i) {
        int num;
        cin >> num;
        cout << reverseNumber(num) << " ";
    }

    cout << endl;
    return 0;
}
